rust/blockstore/src/arrow/block/types.rs

```diff
@@ -66,88 +66,119 @@ impl Block {
         delta
     }
 
-    fn binary_search<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+    /// Binary search the blockfile to find the partition point of the specified prefix and key.
+    /// The implementation is based on [`std::slice::partition_point`].
+    ///
+    /// `(prefix, key)` serves as the search key, and it is sorted in ascending order.
+    /// The partition is defined by: `|x| x < (prefix, key)`.
+    /// The code is a result of inlining this predicate in [`std::slice::partition_point`].
+    /// If the key is unspecified (i.e. [`None`]), we find the first index of the prefix.
+    #[inline]
+    fn binary_search_index<'me, K: ArrowReadableKey<'me>>(
```

Comment:
nit: The coupling of key=None and key=Some cases together in the same api seems forced to me

Reply:
In our case `None` is equivalent to `K::MIN`, but we do not have a `Min` trait so an `Option<K>` is the best alternative I can think of.